p2p/5.Create_an_ERC-20/README.md

```diff
@@ -0,0 +1,245 @@
+
+- Initialize `_totalSupply` in the [constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor) of the contract.
+> ⚠️ Don't forget to assign `_totalSupply` to a wallet and to emit an event
+
+### 📚 **Documentation**:
+
+- [Variable visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility)
+- [Mapping](https://docs.soliditylang.org/en/v0.8.21/types.html#mapping-types)
+- [Constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor)
+
+## Step 3 - Let's get into the hard
+
+### 📑 **Description**:
+
+In this step, you will do the most important part: the implementation of the standard functions.
+
+### 📌 **Tasks**:
+
+- Implement in the `ERC20` contract all the functions present in the `IERC20` interface.
+    - You can copy function prototypes directly into the contract.
+
+> ⚠️ Don't forget to add the `override` modifier
+
+### 👨‍🏫 **Advices**:
+
+- Read the doc of the [ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20), it will help you to understand the purpose of the different functions.
+- Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your functions!
+- Use if into revert or require, to handle [error cases](https://docs.soliditylang.org/en/v0.8.21/contracts.html#errors-and-the-revert-statement).
+
+### 📚 **Documentation**:
+
+- [ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20)
+- [Function visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#function-visibility)
+- [Error cases](https://docs.soliditylang.org/en/v0.8.21/contracts.html#errors-and-the-revert-statement)
+
+## Step 4 - Let's test it
+
+### 📑 **Description**:
+
+In this step, you will test, with foundry, the functions that you have implemented previously.
+
+### 📌 **Tasks**:
+
+- Copy the `ERC20.t.sol` file which is in the `utils` folder to the `test` folder.
+    - This file contains the [foundry tests](https://book.getfoundry.sh/forge/fuzz-testing) that allow you to test all the functions of an ERC-20.
+- Execute the command `forge test` to run the tests.
+    - You can use `forge test -vvvv` to show more informations.
+
+### 📚 **Documentation**:
+
+- [Foundry tests](https://book.getfoundry.sh/forge/fuzz-testing)
+
+### ✔️ **Validation**:
+
+If you have correctly implemented the functions, you should have something like this:
+> ⚠️ To validate a test, you must validate the previous one
+
+```bash
+Running 5 tests for test/ERC20.t.sol:ERC20Test
+[PASS] testApproveAndAllowance() (gas: 31150)
+[PASS] testBalanceOf() (gas: 11065)
+[PASS] testTotalSupply() (gas: 10862)
+[PASS] testTranferFrom() (gas: 49405)
+[PASS] testTransfer() (gas: 36491)
+Test result: ok. 5 passed; 0 failed; finished in 1.72ms
+```
+
+If not, fix the function implementation.
+
+## BONUS Step 5 - Let's deploy it
```

Comment:
you can remove the "bonus" title and add a few bonuses idea later just before the conclusion in the form of a bullet point list for people who want to learn more